src/main/java/org/apache/maven/plugin/resources/remote/AbstractProcessRemoteResourcesMojo.java

```diff
@@ -0,0 +1,1358 @@
+            // check if the artifact is part of the reactor
+            if ( mavenSession != null )
+            {
+                List<MavenProject> list = mavenSession.getProjects();
+                for ( MavenProject p : list )
+                {
+                    if ( s[0].equals( p.getGroupId() ) && s[1].equals( p.getArtifactId() )
+                            && s[2].equals( p.getVersion() ) )
+                    {
+                        if ( s.length >= 4 && "test-jar".equals( s[3] ) )
+                        {
+                            artifactFile = new File( p.getBuild().getTestOutputDirectory() );
+                        }
+                        else
+                        {
+                            artifactFile = new File( p.getBuild().getOutputDirectory() );
+                        }
+                    }
+                }
+            }
+            if ( artifactFile == null || !artifactFile.exists() )
+            {
+                String g = s[0];
+                String a = s[1];
+                String v = s[2];
+                String type = ( s.length >= 4 ? s[3] : "jar" );
+                ArtifactType artifactType = RepositoryUtils.newArtifactType( type,
+                        artifactHandlerManager.getArtifactHandler( type ) );
+                String classifier = ( s.length == 5 ? s[4] : artifactType.getClassifier() );
+
+                DefaultArtifact artifact = new DefaultArtifact(
+                        g, a, classifier, artifactType.getExtension(), v, artifactType );
+
+                try
+                {
+                    ArtifactRequest request = new ArtifactRequest(
+                            artifact,
+                            RepositoryUtils.toRepos( project.getRemoteArtifactRepositories() ),
+                            "remote-resources" );
+                    ArtifactResult result = repositorySystem.resolveArtifact(
+                            mavenSession.getRepositorySession(), request );
+                    artifactFile = result.getArtifact().getFile();
+                }
+                catch ( ArtifactResolutionException e )
+                {
+                    throw new MojoExecutionException( "Error processing remote resources", e );
+                }
+
+            }
+            bundleArtifacts.add( artifactFile );
+        }
+
+        return bundleArtifacts;
+    }
+
+    private ClassLoader initalizeClassloader( List<File> artifacts )
+            throws MojoExecutionException
+    {
+        RemoteResourcesClassLoader cl = new RemoteResourcesClassLoader( null );
+        try
+        {
+            for ( File artifact : artifacts )
+            {
+                cl.addURL( artifact.toURI().toURL() );
+            }
+            return cl;
+        }
+        catch ( MalformedURLException e )
+        {
+            throw new MojoExecutionException( "Unable to configure resources classloader: " + e.getMessage(), e );
```

Comment:
Duplicate message

Reply:
This is how it is in original code, did not touch this area.